### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by sklam**
* remove old API: `autojit`.
* fix very old examples that were broken
* make all examples test-able:
    * `python runtests.py examples.tests`.
    * make code coverage work.

### CODE REVIEW

File: examples/example.py [hunk 1]
```diff
@@ -2,31 +2,11 @@
 # -*- coding: utf-8 -*-
 from __future__ import print_function, division, absolute_import
 
-from scipy.misc import lena
+from scipy.misc import ascent
```
Comment (pitrou) [inline examples/example.py]: Perhaps rename this file ("example.py") to something more descriptive?

File: examples/mandel/mandel_autojit.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
 #! /usr/bin/env python
 # -*- coding: utf-8 -*-
 from __future__ import print_function, division, absolute_import
-from numba import autojit
+
```
Comment (pitrou) [inline examples/mandel/mandel_autojit.py]: Is it worth keeping both this file and `mandel.py`?

**Comment by pitrou** [comment]
The doc failed building following the rename: see https://travis-ci.org/numba/numba/jobs/160189904#L643

**Comment by sklam** [comment]
Fixed

**Pull request merged**